### ISSUE TITLE
sms not working with U of I Shibboleth IdP

### ISSUE THREAD

**Issue by ddriddle** [techservicesillinois/awscli-login#67]
@cmaturi and I are not able to retrieve SMS codes when using the `sms` factor. @kwessel looked at the IdP server logs and found:

```
2021-02-04 14:21:17,977 - WARN [net.shibboleth.idp.authn.duo.impl.ExtractDuoAuthenticationFromHeaders:205] - Profile Action ExtractDuoAuthenticationFromHeaders: Request for SMS codes unsupported
```

@kwessel has sent an email to the Shibboleth user's group to determine if dropping support for SMS was in error or not. If there is no plan to restore support for SMS then mention of SMS needs to be removed from the documentation, and the code should throw an error if `sms` is supplied as a factor.

**Comment by ddriddle** [comment]
Update from @kwessel:

> Looks liks this is, in deed, failing by design because Scott didn't realize that Duo would still send the SMS code even though the auti API returned an error to the user. OSU apparently has a web portal where folks can go to have the SMS code sent. That's a cleaner user experience, but it's two places for the user to go, and it's an extra application to support. I'm personally okay with it failing authentication but sending the SMS code, and I'm filing a feature request to remove the SMS check and bomb when it sees that value.
> 
> I'll let you know what I learn about getting access to Duo device IDs.
> 
> Keith
> -----Original Message-----
> From: users <users-bounces@shibboleth.net> On Behalf Of Cantor, Scott
> Sent: Thursday, February 4, 2021 3:03 PM
> To: Shib Users <users@shibboleth.net>
> Subject: Re: Duo and SMS not supported
>

**Comment by ddriddle** [comment]
Another email update from @kwessel:

> The Duo SMS issue in Shib has been addressed, and the fix should be released in IdP 4.1 which should be out next month.
